### DISCUSSION TITLE
Error 403 when pushing repository

### DISCUSSION THREAD

**Discussion by derektypist** [community/community#32006] [Other Feature Feedback, Questions, & Ideas]
I am getting error code 403 when pushing a repository.

I get the message 

remote: Permission to `username/projectname.git` denied to username.
fatal: unable to access `https://github.com/username/projectname.git`: The requested URL returned error: 403.

What action should I take, if I get this error message?

**Comment by airtower-luna** [comment]
Either your token doesn't have the right scopes (**repo**, plus **workflow** if you use Actions), or you don't have write access to the repository.

In the former case you need to generate a new token and use that for pushing, in the latter you need to either (depending on the desired workflow) create a fork, push there, and make a PR, or talk to a repository owner/admin so they allow you access.
